components/org.wso2.carbon.identity.application.authenticator.samlsso/src/test/java/org/wso2/carbon/identity/application/authenticator/samlsso/SAMLSSOAuthenticatorTest.java

```diff
@@ -0,0 +1,236 @@
+        when(mockedAuthenticationContext.getAuthenticatorProperties()).thenReturn(authenticatorProperties);
+        when(mockedAuthenticationContext.getContextIdentifier()).thenReturn(TestConstants.RELAY_STATE);
+
+        when(mockedExternalIdPConfig.getIdentityProvider()).thenReturn(mockedIdentityProvider);
+        when(mockedAuthenticationContext.getExternalIdP()).thenReturn(mockedExternalIdPConfig);
+
+        when(mockedAuthenticationRequest.getRequestQueryParam(SSOConstants.HTTP_POST_PARAM_SAML2_AUTH_REQ))
+                .thenReturn(new String[]{TestConstants.SAML2_REDIRECT_REQUEST});
+        when(mockedAuthenticationRequest.isPost()).thenReturn(Boolean.FALSE);
+        when(mockedAuthenticationContext.getAuthenticationRequest()).thenReturn(mockedAuthenticationRequest);
+
+        samlssoAuthenticator.initiateAuthenticationRequest(mockedHttpServletRequest, mockedAuthnHttpServletResponse,
+                mockedAuthenticationContext);
+
+        ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+        verify(mockedAuthnHttpServletResponse).sendRedirect(captor.capture());
+        Assert.assertTrue(captor.getValue().contains("SAMLRequest"));
+        Assert.assertTrue(captor.getValue().contains("RelayState"));
+    }
+
+    @Test(priority = 5)
+    public void testInitiatePostLogoutRequest() throws LogoutFailedException, IOException {
+
+        Map<String, String> authenticatorProperties = new HashMap<>();
+        authenticatorProperties.put(IdentityApplicationConstants.Authenticator.SAML2SSO.SSO_URL,
+                "https://localhost:9443/samlsso");
+        authenticatorProperties.put(IdentityApplicationConstants.Authenticator.SAML2SSO.REQUEST_METHOD,
+                SSOConstants.POST);
+        authenticatorProperties.put(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_LOGOUT_ENABLED, "true");
+        when(mockedAuthenticationContext.getAuthenticatorProperties()).thenReturn(authenticatorProperties);
+
+        when(mockedHttpSession.getAttribute(SSOConstants.LOGOUT_USERNAME)).thenReturn("SomeUser");
+        when(mockedHttpSession.getAttribute(SSOConstants.LOGOUT_SESSION_INDEX)).thenReturn("123456789");
+        when(mockedHttpSession.getAttribute(SSOConstants.NAME_QUALIFIER)).thenReturn(NameIDType.UNSPECIFIED);
+        when(mockedHttpSession.getAttribute(SSOConstants.SP_NAME_QUALIFIER)).thenReturn(NameIDType.UNSPECIFIED);
+        when(mockedHttpServletRequest.getSession()).thenReturn(mockedHttpSession);
+
+        PrintWriter out = new PrintWriter("logout-request.txt");
+        when(mockedLocoutHttpServletResponse.getWriter()).thenReturn(out);
+
+        samlssoAuthenticator.initiateLogoutRequest(mockedHttpServletRequest, mockedLocoutHttpServletResponse,
+                mockedAuthenticationContext);
+        out.flush();
+        String postPage = FileUtils.readFileToString(new File("logout-request.txt"), "UTF-8");
+        Assert.assertTrue(postPage.contains("SAMLRequest"));
```

Comment:
Shall we add a message here ? Otherwise if this fails it will only show expected true but was false.